### PR TITLE
CSEN-19 JsonMapper was implemented

### PR DESCRIPTION
**Pull request by MedvedevBW**

### CODE REVIEW

**Comment by MedvedevBW** [comment]
@placccebo

File: src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+package com.bwsw.cloudstack.entities.common
+
+import com.fasterxml.jackson.databind.JsonMappingException
+import org.scalatest.{FlatSpec, PrivateMethodTester}
+
+class JsonMapperTestSuite extends FlatSpec with PrivateMethodTester {
+  val firstValue = "1"
+  val secondValue = "2"
+
+  "deserialize" should "deserialize json string to simple case class if setIgnoreUnknownProperties flag is true" +
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala]: deserialize should convert a json string to an entity if this string contains the same fields as the entity class and setIgnoreUnknownProperties flag is true

Comment (placccebo) [reply src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala]: change the rest accordingly

File: src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  "deserialize" should "throw JsonMappingException if setIgnoreUnknownProperties flag is false" +
+    "and Json string has non equals fields with the target entity" in {
+    val testJson = "{\"first\":\"" + s"$firstValue" + "\",\"second\":\"" + s"$secondValue" + "\",\"third\":\"3\"}"
+    val expectedEntity = Test.TestEntity(firstValue, secondValue)
+    val jsonMapper = new JsonMapper(false)
+
+    assertThrows[JsonMappingException](jsonMapper.deserialize[Test.TestEntity](testJson))
+  }
+
+  "serialize" should "serialize entity to Json string" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala]: "serialize" should "convert an entity to a json string"

File: src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    val testJson = "{\"first\":\"" + s"$firstValue" + "\",\"second\":\"" + s"$secondValue" + "\"}"
+    val jsonMapper = new JsonMapper()
+
+    assert(jsonMapper.serialize(Test.TestEntity(firstValue, secondValue)) == testJson)
+  }
+
+  "getIgnoreUnknownPropertiesFlag" should "get value of IgnoreUnknownProperties flag" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala]: "getIgnoreUnknownPropertiesFlag" should "return a value of IgnoreUnknownProperties flag"

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala (hunk 1 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala (hunk 2 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala (hunk 3 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/entities/common/JsonMapperTestSuite.scala]: ok

**Review by placccebo — APPROVED** [review]

**Pull request merged**